### PR TITLE
App-project: Persist user sessions in session storage

### LINKED ISSUE #2980: Logged-in status isn't remembered across page visits, during a session
## Package
app-project

## Describe the bug
When I leave the project app, to check Talk or my messages, then return, I have to wait to be logged in again, even if I already have a valid Panoptes session.

https://user-images.githubusercontent.com/59547/159287692-74bf4ff0-b012-4dc0-a5f9-0de75dcf838f.mov

## To Reproduce
1. Visit a page on an FEM project, while logged-in eg. https://frontend.preview.zooniverse.org/projects/nora-dot-eisner/planet-hunters-tess/classify/workflow/11235
2. Wait to be authenticated.
3. Go to Talk and return, or just reload the page.
4. Wait to be authenticated again, before the classifier can load.

## Expected behavior
User logged-in status (`store.user.loadingState`) should be remembered across visits during a browser session.

### PR DESCRIPTION
**Pull request by eatyourgreens**
- Save user snapshots to `window.sessionStorage`.
- When a project page first loads in the browser, read user data from session storage and render the page instantly.
- Add a `usePanoptesUser` hook to fetch up-to-date user data in the background.
- Remove `checkCurrent` from the `User` model.

With the changes here, you should see much faster loading times when leaving the project app and returning, eg. going to Talk or your inbox, then returning to the Classify page.

https://user-images.githubusercontent.com/59547/159286530-6407cd0b-9028-4e30-85aa-9ff5fca6c417.mov

Testing checklist (all tested locally on PH-TESS.)
- [x] Log in and log out still work within the project app.
- [x] Logging out outside the project app also logs me out when I enter a project page.
- [x] Logging in outside the project app also logs me in when I enter a project page.
 
Package:
app-project

Closes #2980.

# Review Checklist

## General

- [x] Are the tests passing locally and on Travis?
- [x] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [x] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

**Comment by eatyourgreens** [comment]
A `useProjectRoles` hook (see #2940) would be fairly quick to build on top of the work here. That would enable us to add support for admin features, for example (see https://github.com/zooniverse/front-end-monorepo/issues/623.)

**Review by mcbouslog — APPROVED** [review]
LGTM!

Tested signing in, signing out, starting from zooniverse.org and TESS, going back and forth (comparing zooniverse.org or 404 page locally to TESS in FEM locally and production). Everything worked as expected, including user stats loaded and made classifications.

File: packages/app-project/package.json [hunk 1]
```diff
@@ -63,6 +63,7 @@
     "react-resize-detector": "~7.1.2",
     "styled-components": "~5.3.3",
     "svg-loaders-react": "~2.2.1",
+    "swr": "~1.2.2",
```
Comment (eatyourgreens) [inline packages/app-project/package.json:66]: This is out of date now. I should bump to the latest version before merging.

File: packages/app-project/src/hooks/usePanoptesUser.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import auth from 'panoptes-client/lib/auth'
+import useSWR from 'swr'
+
+const SWRoptions = {
+  revalidateIfStale: true,
+  revalidateOnMount: true,
+  revalidateOnFocus: true,
+  revalidateOnReconnect: true,
+  refreshInterval: 0
```
[the remark this answers is not in the document]
Comment (eatyourgreens) [reply packages/app-project/src/hooks/usePanoptesUser.js:9]: Yeah, these are the default options from https://swr.vercel.app/docs/options#options

**Pull request merged**